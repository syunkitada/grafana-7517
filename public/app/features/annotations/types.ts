import { PanelData, DataFrame, AnnotationEvent, TimeRange } from '@grafana/data';
import { DashboardModel, PanelModel } from '../dashboard/state';

export interface AnnotationQueryOptions {
  dashboard: DashboardModel;
  panel: PanelModel;
  range: TimeRange;
}

export interface AnnotationQueryResponse {
  /**
   * All the data flattened to a single frame
   */
  frame?: DataFrame;

  /**
   * The processed annotation events
   */
  events?: AnnotationEvent[];

  /**
   * The original panel response
   */
  panelData?: PanelData;
}
