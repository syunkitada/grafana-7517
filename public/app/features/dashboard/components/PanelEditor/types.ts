import { PanelModel } from '../../state/PanelModel';

export enum DisplayMode {
  Fill = 0,
  Fit = 1,
  Exact = 2,
}

export const displayModes = [
  { value: DisplayMode.Fill, label: 'Fill', description: 'Use all avaliable space' },
  { value: DisplayMode.Fit, label: 'Fit', description: 'Fit in the space keeping ratio' },
  { value: DisplayMode.Exact, label: 'Exact', description: 'Same size as the dashboard' },
];

export enum EditorTab {
  Query = 'query',
  Alerts = 'alerts',
  Transform = 'xform',
}

export const allTabs = [
  { tab: EditorTab.Query, label: 'Query', show: (panel: PanelModel) => true },
  { tab: EditorTab.Alerts, label: 'Alerts', show: (panel: PanelModel) => true },
  { tab: EditorTab.Transform, label: 'Transform', show: (panel: PanelModel) => true },
];
