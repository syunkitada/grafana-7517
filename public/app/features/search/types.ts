export enum DashboardSearchItemType {
  DashDB = 'dash-db',
  DashHome = 'dash-home',
  DashFolder = 'dash-folder',
}

export interface DashboardSection {
  id: number;
  uid?: string;
  title: string;
  expanded?: boolean;
  url: string;
  icon: string;
  score: number;
  hideHeader?: boolean;
  checked?: boolean;
  items: DashboardSectionItem[];
  toggle?: (section: DashboardSection) => Promise<DashboardSection>;
  selected?: boolean;
  type: DashboardSearchItemType;
}

export interface DashboardSectionItem {
  checked?: boolean;
  folderId?: number;
  folderTitle?: string;
  folderUid?: string;
  folderUrl?: string;
  id: number;
  isStarred: boolean;
  selected?: boolean;
  tags: string[];
  title: string;
  type: DashboardSearchItemType;
  uid: string;
  uri: string;
  url: string;
}

export interface DashboardTag {
  term: string;
  count: number;
}

export interface DashboardQuery {
  query: string;
  mode: string;
  tag: string[];
  starred: boolean;
  skipRecent: boolean;
  skipStarred: boolean;
  folderIds: number[];
}

export interface SectionsState {
  sections: DashboardSection[];
  allChecked: boolean;
  dashboardTags: DashboardTag[];
}

export type ItemClickWithEvent = (item: DashboardSectionItem | DashboardSection, event: any) => void;

export type SearchAction = {
  type: string;
  payload?: any;
};

export interface OpenSearchParams {
  query?: string;
}
