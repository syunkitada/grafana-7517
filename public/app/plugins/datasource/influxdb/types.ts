import { DataQuery, DataSourceJsonData } from '@grafana/data';

export interface InfluxOptions extends DataSourceJsonData {
  timeInterval: string;
  httpMode: string;

  // Influx 2.0
  enableFlux?: boolean;
  organization?: string;
  defaultBucket?: string;
  maxSeries?: number;
}

export interface InfluxSecureJsonData {
  password?: string;
  token?: string;
}

export interface InfluxQueryPart {
  type: string;
  params?: string[];
  interval?: string;
}

export interface InfluxQueryTag {
  key: string;
  operator?: string;
  condition?: string;
  value: string;
}

export enum InfluxQueryType {
  Classic = 'Classic', // IFQL query builder
  InfluxQL = 'InfluxQL', // raw ifql
  Flux = 'Flux',
}

export interface InfluxQuery extends DataQuery {
  queryType?: InfluxQueryType;
  policy?: string;
  measurement?: string;
  resultFormat?: 'time_series' | 'table';
  orderByTime?: string;
  tags?: InfluxQueryTag[];
  groupBy?: InfluxQueryPart[];
  select?: InfluxQueryPart[][];
  limit?: string;
  slimit?: string;
  tz?: string;
  fill?: string;
  rawQuery?: boolean; // deprecated (use raw InfluxQL)
  query?: string;
}
