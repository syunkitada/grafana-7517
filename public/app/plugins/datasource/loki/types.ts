import { DataQuery } from '@grafana/ui/src/types';

export interface LokiQuery extends DataQuery {
  expr: string;
}

