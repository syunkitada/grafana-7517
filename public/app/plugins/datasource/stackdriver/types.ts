export enum MetricFindQueryTypes {
  Services = 'services',
  MetricTypes = 'metricTypes',
  MetricLabels = 'metricLabels',
  ResourceLabels = 'resourceLabels',
  ResourceTypes = 'resourceTypes',
  Aggregations = 'aggregations',
  Alignerns = 'alignerns',
  AlignmentPeriods = 'alignmentPeriods',
}

export interface TemplateQueryComponentData {
  selectedQueryType: string;
  metricDescriptors: any[];
  selectedService: string;
  selectedMetricType: string;
  labels: string[];
  labelKey: string;
  metricTypes: Array<{ value: string; name: string }>;
  services: Array<{ value: string; name: string }>;
}
