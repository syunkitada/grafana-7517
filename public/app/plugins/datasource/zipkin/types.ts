export type ZipkinSpan = {
  traceId: string;
  parentId?: string;
  name: string;
  id: string;
  timestamp: number;
  duration: number;
  localEndpoint: {
    serviceName: string;
    ipv4: string;
    port?: number;
  };
  annotations?: ZipkinAnnotation[];
  tags?: { [key: string]: string };
  kind?: 'CLIENT' | 'SERVER' | 'PRODUCER' | 'CONSUMER';
};

export type ZipkinAnnotation = {
  timestamp: number;
  value: string;
};
