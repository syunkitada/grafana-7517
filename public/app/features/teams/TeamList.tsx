import React, { PureComponent } from 'react';
import { connect } from 'react-redux';
import { hot } from 'react-hot-loader';
import PageHeader from 'app/core/components/PageHeader/PageHeader';
import DeleteButton from 'app/core/components/DeleteButton/DeleteButton';
import { NavModel, Team } from '../../types';
import { loadTeams } from './state/actions';
import { getTeams } from './state/selectors';
import { getNavModel } from 'app/core/selectors/navModel';

export interface Props {
  navModel: NavModel;
  teams: Team[];
  loadTeams: typeof loadTeams;
  search: string;
}

export class TeamList extends PureComponent<Props, any> {
  componentDidMount() {
    this.fetchTeams();
  }

  async fetchTeams() {
    await this.props.loadTeams();
  }

  deleteTeam = (team: Team) => {
    console.log('delete team', team);
  };

  onSearchQueryChange = event => {
    console.log('set search', event.target.value);
  };

  renderTeamMember(team: Team) {
    const teamUrl = `org/teams/edit/${team.id}`;

    return (
      <tr key={team.id}>
        <td className="width-4 text-center link-td">
          <a href={teamUrl}>
            <img className="filter-table__avatar" src={team.avatarUrl} />
          </a>
        </td>
        <td className="link-td">
          <a href={teamUrl}>{team.name}</a>
        </td>
        <td className="link-td">
          <a href={teamUrl}>{team.email}</a>
        </td>
        <td className="link-td">
          <a href={teamUrl}>{team.memberCount}</a>
        </td>
        <td className="text-right">
          <DeleteButton onConfirmDelete={() => this.deleteTeam(team)} />
        </td>
      </tr>
    );
  }

  render() {
    const { navModel, teams, search } = this.props;

    return (
      <div>
        <PageHeader model={navModel as NavModel} />
        <div className="page-container page-body">
          <div className="page-action-bar">
            <div className="gf-form gf-form--grow">
              <label className="gf-form--has-input-icon gf-form--grow">
                <input
                  type="text"
                  className="gf-form-input"
                  placeholder="Search teams"
                  value={search}
                  onChange={this.onSearchQueryChange}
                />
                <i className="gf-form-input-icon fa fa-search" />
              </label>
            </div>

            <div className="page-action-bar__spacer" />

            <a className="btn btn-success" href="org/teams/new">
              <i className="fa fa-plus" /> New team
            </a>
          </div>

          <div className="admin-list-table">
            <table className="filter-table filter-table--hover form-inline">
              <thead>
                <tr>
                  <th />
                  <th>Name</th>
                  <th>Email</th>
                  <th>Members</th>
                  <th style={{ width: '1%' }} />
                </tr>
              </thead>
              <tbody>{teams.map(team => this.renderTeamMember(team))}</tbody>
            </table>
          </div>
        </div>
      </div>
    );
  }
}

function mapStateToProps(state) {
  return {
    navModel: getNavModel(state.navIndex, 'teams'),
    teams: getTeams(state.teams),
    search: '',
  };
}

const mapDispatchToProps = {
  loadTeams,
};

export default hot(module)(connect(mapStateToProps, mapDispatchToProps)(TeamList));
