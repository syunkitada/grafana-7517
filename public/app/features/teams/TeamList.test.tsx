import React from 'react';
import { shallow } from 'enzyme';
import { TeamList, Props } from './TeamList';
import { NavModel, Team } from '../../types';

const setup = (propOverrides?: object) => {
  const props: Props = {
    navModel: {} as NavModel,
    teams: [] as Team[],
    loadTeams: jest.fn(),
    search: '',
  };

  Object.assign(props, propOverrides);

  const wrapper = shallow(<TeamList {...props} />);
  const instance = wrapper.instance() as TeamList;

  return {
    wrapper,
    instance,
  };
};

describe('Render', () => {
  it('should render component', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('should render teams table', () => {
    const { wrapper } = setup({
      teams: [
        {
          id: 1,
          name: 'test',
          avatarUrl: 'some/url/',
          email: 'test@test.com',
          memberCount: 1,
          search: '',
          members: [],
          groups: [],
        },
      ],
    });

    expect(wrapper).toMatchSnapshot();
  });
});

describe('Life cycle', () => {
  it('should call loadTeams', () => {
    const { instance } = setup();

    instance.componentDidMount();

    expect(instance.props.loadTeams).toHaveBeenCalled();
  });
});

describe('Functions', () => {});
