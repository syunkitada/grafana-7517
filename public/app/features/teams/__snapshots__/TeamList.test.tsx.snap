// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render should render component 1`] = `
<div>
  <PageHeader
    model={Object {}}
  />
  <div
    className="page-container page-body"
  >
    <div
      className="page-action-bar"
    >
      <div
        className="gf-form gf-form--grow"
      >
        <label
          className="gf-form--has-input-icon gf-form--grow"
        >
          <input
            className="gf-form-input"
            onChange={[Function]}
            placeholder="Search teams"
            type="text"
            value=""
          />
          <i
            className="gf-form-input-icon fa fa-search"
          />
        </label>
      </div>
      <div
        className="page-action-bar__spacer"
      />
      <a
        className="btn btn-success"
        href="org/teams/new"
      >
        <i
          className="fa fa-plus"
        />
         New team
      </a>
    </div>
    <div
      className="admin-list-table"
    >
      <table
        className="filter-table filter-table--hover form-inline"
      >
        <thead>
          <tr>
            <th />
            <th>
              Name
            </th>
            <th>
              Email
            </th>
            <th>
              Members
            </th>
            <th
              style={
                Object {
                  "width": "1%",
                }
              }
            />
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
  </div>
</div>
`;

exports[`Render should render teams table 1`] = `
<div>
  <PageHeader
    model={Object {}}
  />
  <div
    className="page-container page-body"
  >
    <div
      className="page-action-bar"
    >
      <div
        className="gf-form gf-form--grow"
      >
        <label
          className="gf-form--has-input-icon gf-form--grow"
        >
          <input
            className="gf-form-input"
            onChange={[Function]}
            placeholder="Search teams"
            type="text"
            value=""
          />
          <i
            className="gf-form-input-icon fa fa-search"
          />
        </label>
      </div>
      <div
        className="page-action-bar__spacer"
      />
      <a
        className="btn btn-success"
        href="org/teams/new"
      >
        <i
          className="fa fa-plus"
        />
         New team
      </a>
    </div>
    <div
      className="admin-list-table"
    >
      <table
        className="filter-table filter-table--hover form-inline"
      >
        <thead>
          <tr>
            <th />
            <th>
              Name
            </th>
            <th>
              Email
            </th>
            <th>
              Members
            </th>
            <th
              style={
                Object {
                  "width": "1%",
                }
              }
            />
          </tr>
        </thead>
        <tbody>
          <tr
            key="1"
          >
            <td
              className="width-4 text-center link-td"
            >
              <a
                href="org/teams/edit/1"
              >
                <img
                  className="filter-table__avatar"
                  src="some/url/"
                />
              </a>
            </td>
            <td
              className="link-td"
            >
              <a
                href="org/teams/edit/1"
              >
                test
              </a>
            </td>
            <td
              className="link-td"
            >
              <a
                href="org/teams/edit/1"
              >
                test@test.com
              </a>
            </td>
            <td
              className="link-td"
            >
              <a
                href="org/teams/edit/1"
              >
                1
              </a>
            </td>
            <td
              className="text-right"
            >
              <DeleteButton
                onConfirmDelete={[Function]}
              />
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
